### PR TITLE
is exist check correction

### PR DESCRIPTION
**Pull request by rcohencyberarmor**

### CODE REVIEW

**Review by Daniel-GrunbergerCA — CHANGES_REQUESTED** [review]
Add tests

**Comment by rcohencyberarmor** [comment]
there are tests

**Comment by Daniel-GrunbergerCA** [comment]
yes but it didn't cover your case :)

File: cloudsupport/v1/cloudproviderv1.go [hunk 1]
```diff
@@ -329,7 +329,8 @@ func GetPolicyVersionAKS(aksSupport IAKSSupport, cluster string, subscriptionId
 // check if the server is AKS. e.g. https://XXX.XX.XXX.azmk8s.io:443
 func IsAKS() bool {
 	const serverIdentifierAKS = "azmk8s.io"
-	clusterServerName := k8sinterface.GetK8sConfigClusterServerName()
+	config := k8sinterface.GetConfig()
```
Comment (Daniel-GrunbergerCA) [inline cloudsupport/v1/cloudproviderv1.go:332]: what if this is `nil`?

File: k8sinterface/k8sconfig.go [hunk 2]
```diff
@@ -186,10 +186,9 @@ func SetConfigClusterServerName(contextName string) {
 	ConfigClusterServerName = contextName
 }
 
-func GetK8sConfigClusterServerName() string {
+func GetK8sConfigClusterServerName(config *clientcmdapi.Config) string {
 	if ConfigClusterServerName == "" {
-		config := GetConfig()
-		if _, exist := config.Clusters[config.CurrentContext]; !exist {
+		if _, exist := config.Clusters[config.CurrentContext]; exist {
```
Comment (Daniel-GrunbergerCA) [inline k8sinterface/k8sconfig.go:191]: add a test that if tomorrow someone changes `exist` to `!exist` it will fail, since this was the original bug

**Review by Daniel-GrunbergerCA — CHANGES_REQUESTED** [review]

**Pull request merged**